Towny/src/main/java/com/palmergames/bukkit/util/ItemLists.java

```diff
@@ -278,11 +282,49 @@ public Collection<String> getMaterialNameCollection() {
 			.includeList(PLANTABLES)
 			.add("TALL_GRASS","LARGE_FERN","VINE","TWISTING_VINES_PLANT","WEEPING_VINES_PLANT","NETHER_WART_BLOCK","CRIMSON_ROOTS","WARPED_ROOTS","NETHER_SPROUTS","BIG_DRIPLEAF","SMALL_DRIPLEAF").build();
 
+	/**
+	 * Minecraft uses 3 types of air
+	 */
+	public static final ItemLists AIR_TYPES = newBuilder()
+		.add("AIR")
+		.add("CAVE_AIR")
+		.add("VOID_AIR").build();
+
+	/**
+	 * List of all the banners
+	 */
+	public static final ItemLists BANNERS = newBuilder().endsWith("_BANNER").build();
+
+	/**
+	 * List of wall coral fans, players can fall through these
+	 */
+	public static final ItemLists CORAL_FANS = newBuilder()
+		.endsWith("_WALL_FAN")
+		.build();
+		
+	/**
+	 * List of solid blocks
+	 */
+	public static final ItemLists NOT_SOLID_BLOCKS = newBuilder()

```

Comment:
I don't see how this is maintainable in the long run, I don't want to have to add new blocks to this every version update

Reply:
Paper has an isCollidable method that is better suited for checking what we typically consider solid, this can be used with some reflection (See JavaUtil#getMethodHandle) which also makes this item list redundant.

Reply:
I cannot find the `isCollidable()` method you mentioned, the closest thing I could find in the Documentation is `LivingEntity#isCollidable()` which has nothing to do with this use case.

Not sure what you mean how using reflection could help here

Reply:
I agree on the maintainability of this. Some of the buttons/plates/gates/doors will get updated but you'll be tasked with maintaining any other blocks here.

Reply:
I know it might be vexing to mantain, however I didn't come up with a better solution, I even tought of using `Material#isOccluding()` however that comes with its set of problems too like solid blocks that are transparent like glass, glowstone and more, and would be even worse to mantain